services/user.js

```diff
@@ -803,7 +861,144 @@ class DataCommon {
 }
 
 
+class UserActionPermissions {
+    constructor(role) {
+        this.role = role;
+    }
+
+    static get(role, permissions) {
+        const userPermissions = new UserActionPermissions(role);
+        return userPermissions.#getPermissions(permissions);
+    }
+
+    static getNewUserPermission(role) {
+        const permissions = new UserActionPermissions(role);
+        return permissions.#getDefaultPermissionsByRole();
+    }
+
+    #getPermissions(permissions) {
+        const editableUserRoles = [USER.ROLES.FEDERAL_LEAD, USER.ROLES.FEDERAL_MONITOR, USER.ROLES.DATA_COMMONS_PERSONNEL, USER.ROLES.CURATOR, USER.ROLES.DC_POC];
+        const defaultPermissions = UserActionPermissions.getNewUserPermission(this.role);
+        if (editableUserRoles.includes(this.role)) {
+            const bannedPermissions = this.#getBannedPermissionsByRole(this.role);
+            const bannedPermissionsSet = new Set(bannedPermissions);
+            return permissions ? permissions.filter(permission => !bannedPermissionsSet.has(permission)) : defaultPermissions;
+        }
+        return defaultPermissions;
+    }
+
+    #getDefaultPermissionsByRole() {
+        switch (this.role) {
+            case USER.ROLES.FEDERAL_LEAD:
+                return [SUBMISSION_REQUEST.VIEW, ADMIN.VIEW_DASHBOARD, DATA_SUBMISSION.VIEW];
+            case USER.ROLES.DATA_COMMONS_PERSONNEL:
+                return [DATA_SUBMISSION.VIEW, DATA_SUBMISSION.REVIEW, DATA_SUBMISSION.CONFIRM];
+            case USER.ROLES.ADMIN:
+                return [SUBMISSION_REQUEST.VIEW, DATA_SUBMISSION.VIEW, DATA_SUBMISSION.REVIEW, DATA_SUBMISSION.ADMIN_SUBMIT, DATA_SUBMISSION.CONFIRM,
+                ADMIN.MANAGE_USER, ADMIN.MANAGE_PROGRAMS, ADMIN.MANAGE_STUDIES, ADMIN.VIEW_DASHBOARD];
+            case USER.ROLES.SUBMITTER:
+                return [SUBMISSION_REQUEST.CREATE, DATA_SUBMISSION.REQUEST_ACCESS, DATA_SUBMISSION.VIEW, DATA_SUBMISSION.CREATE];
+            default:
+                // For USER.ROLES.USER:
+                return [SUBMISSION_REQUEST.CREATE, DATA_SUBMISSION.REQUEST_ACCESS];
+        }
+    }
+
+    #getBannedPermissionsByRole() {
+        switch (this.role) {
+            case USER.ROLES.FEDERAL_LEAD:
+                return [DATA_SUBMISSION.ADMIN_SUBMIT, DATA_SUBMISSION.REQUEST_ACCESS];
+            case USER.ROLES.DATA_COMMONS_PERSONNEL:
```

Comment:
There are only 2 banned permissions for Data commons personnel based on the latest spreadsheet for the ticket

Reply:
I've pushed it for the change.